atlasdb-config/src/test/java/com/palantir/atlasdb/factory/TransactionManagersTest.java

```diff
@@ -326,6 +303,90 @@ public void runsClosingCallbackOnShutdown() throws Exception {
         verify(callback, times(1)).run();
     }
 
+    @Test
+    public void metricsAreReportedExactlyOnceWhenUsingLocalService() throws IOException, InterruptedException {
+        setUpForLocalServices();
+        setupLeaderBlockInConfig();
+
+        assertThatTimeAndLockMetricsAreRecorded(TIMESTAMP_SERVICE_FRESH_TIMESTAMP_METRIC, REMOTELOCK_SERVICE_CURRENT_TIME_METRIC);
+    }
+
+    @Test
+    public void metricsAreReportedExactlyOnceWhenUsingRemoteService() throws IOException, InterruptedException {
+        setUpForRemoteServices();
+        setupLeaderBlockInConfig();
+
+        assertThatTimeAndLockMetricsAreRecorded(TIMESTAMP_SERVICE_FRESH_TIMESTAMP_METRIC, REMOTELOCK_SERVICE_CURRENT_TIME_METRIC);
+    }
+
+    @Test
+    public void metricsAreReportedExactlyOnceWhenUsingTimelockService() throws IOException, InterruptedException {
+        when(config.timelock()).thenReturn(Optional.of(mockClientConfig));
+
+        assertThatTimeAndLockMetricsAreRecorded(TIMELOCK_SERVICE_FRESH_TIMESTAMP_METRIC, TIMELOCK_SERVICE_CURRENT_TIME_METRIC);
+    }
+
+    @Test
+    public void metricsAreReportedExactlyOnceWhenUsingTimelockServiceWithRequestBatching() throws IOException, InterruptedException {
+        when(config.timelock()).thenReturn(Optional.of(mockClientConfig));
+        when(runtimeConfig.timestampClient()).thenReturn(ImmutableTimestampClientConfig.of(true));
+
+        assertThatTimeAndLockMetricsAreRecorded(TIMESTAMP_SERVICE_FRESH_TIMESTAMP_METRIC, TIMELOCK_SERVICE_CURRENT_TIME_METRIC);
+
+    }
+
+    private void assertThatTimeAndLockMetricsAreRecorded(String timestampMetric, String lockMetric) {
+        MetricRegistry metrics = MetricRegistries.createWithHdrHistogramReservoirs();
+        AtlasDbMetrics.setMetricRegistry(metrics);
```

Comment:
might want to just use `MetricsRule` to have it handle saving & restoring the metrics

```
    @Rule
    public MetricsRule metricsRule = new MetricsRule();
```